biodatasets/gnormplus/gnormplus.py

```diff
@@ -0,0 +1,236 @@
+
+            for idx, document in enumerate(collection.documents):
+                if self.config.schema == "source":
+                    features = {
+                        "doc_id": document.id,
+                        "passages": [
+                            {
+                                "text": passage.text,
+                                "type": passage.infons["type"],
+                                "location": {"offset": passage.offset, "length": passage.total_span.length},
+                            }
+                            for passage in document.passages
+                        ],
+                        "entities": [
+                            {
+                                "locations": [
+                                    {"offset": loc.offset, "length": loc.length} for loc in entity.locations
+                                ],
+                                "text": entity.text,
+                                "type": entity.infons["type"],
+                                "normalized": [
+                                    {
+                                        "db_id": entity.infons[entity.infons["type"]],
+                                    }
+                                ]
+                                if entity.infons["type"] in entity.infons
+                                else [],
+                            }
+                            for passage in document.passages
+                            for entity in passage.annotations
+                        ],
+                    }
+                    yield idx, features
+                elif self.config.schema == "bigbio_kb":
+
+                    # passage offsets/lengths do not connect, recalculate them for this schema.
+                    passage_spans = []
+                    start = 0
+                    for passage in document.passages:
+                        end = start + len(passage.text)
+                        passage_spans.append((start, end))
+                        start = end + 1
+
+                    features = {
+                        "id": next(uid),
+                        "document_id": document.id,
+                        "passages": [
+                            {
+                                "id": next(uid),
+                                "type": passage.infons["type"],
+                                "text": [passage.text],
+                                "offsets": [span],
+                            }
+                            for passage, span in zip(document.passages, passage_spans)
+                        ],
+                        "entities": [
+                            {
+                                "id": next(uid),
+                                "type": entity.infons["type"],
+                                "text": [entity.text],
+                                "offsets": [(loc.offset, loc.end) for loc in entity.locations],
```

Comment:
Can you please modify this to use `utils.parsing.get_text_and_offsets_from_bioc_ann`?